### PR TITLE
change on CSS 

### PR DESCRIPTION
**Pull request by 117649**
This should make appearance functions like tab tittle font settings and hidden close button on tab work.

### CODE REVIEW

**Comment by onemen** [comment]
Can you look at addon/chrome/skin/app_version
```
chrome.manifest
skin      tabmix-version tabmixplus  chrome/skin/app_version/39.0/       appversion>=39.0a1
skin      tabmix-version tabmixplus  chrome/skin/app_version/60.0/       appversion>=60.0a1
skin      tabmix-os    classic/1.0   chrome/skin/app_version/41.0/win/   os=WINNT appversion>=41.0a1
skin      tabmix-os    classic/1.0   chrome/skin/app_version/60.0/win/   os=WINNT appversion>=60.0a1
skin      tabmix-os    classic/1.0   chrome://tabmix-version/skin/mac/   os=Darwin
skin      tabmix-os    classic/1.0   chrome://tabmix-version/skin/linux/ os=Linux
```

Check if you can combine all the old version folders, i think we can keep the file structure so we can add more folders in the  future as needed.
1. rename app_version/before39.0 -> app_version/86.0
2. remove obsolete rules
3. combine app_version/39.0, app_version/41.0 app_version/60.0 into the files in app_version/86.0
4. update @import in css
5. update chrome.manifest

If all "binding" are done by loading an overly with JavaScript we can remove all  -moz-binding from css

**Comment by 117649** [comment]
> Can you look at addon/chrome/skin/app_version
> 
> ```
> chrome.manifest
> skin      tabmix-version tabmixplus  chrome/skin/app_version/39.0/       appversion>=39.0a1
> skin      tabmix-version tabmixplus  chrome/skin/app_version/60.0/       appversion>=60.0a1
> skin      tabmix-os    classic/1.0   chrome/skin/app_version/41.0/win/   os=WINNT appversion>=41.0a1
> skin      tabmix-os    classic/1.0   chrome/skin/app_version/60.0/win/   os=WINNT appversion>=60.0a1
> skin      tabmix-os    classic/1.0   chrome://tabmix-version/skin/mac/   os=Darwin
> skin      tabmix-os    classic/1.0   chrome://tabmix-version/skin/linux/ os=Linux
> ```
> 
> Check if you can combine all the old version folders, i think we can keep the file structure so we can add more folders in the future as needed.
> 
> 1. rename app_version/before39.0 -> app_version/86.0
> 2. remove obsolete rules
> 3. combine app_version/39.0, app_version/41.0 app_version/60.0 into the files in app_version/86.0
> 4. update @import in css
> 5. update chrome.manifest
> 
> If all "binding" are done by loading an overly with JavaScript we can remove all -moz-binding from css

I think the min version should be 78 which is the latest ESR. 86 hasn't come to my aurora yet..

XBL is no more -moz-binding is useless now. Now you have to do everything by hacking with JavaScript.
This remind me that loading of some .js may should be take out from overlay and forward to bootstrap.js in hope to avoid some problem caused by late load of the script. This is something should be take into consideration in future.

I'll try sort out skin folder first.

By the way my most ahead branch have some fix on problems that blocking drag and drop mechanics from effective and lets say  it just a mess everything is buggy. If you happen to working on that take a look on there first may save you some time.

**Comment by 117649** [comment]
have to rebase this on latest rewrite/main for test.

File: addon/chrome/skin/app_version/78/win/tabStyles.css [hunk 1]
```diff
@@ -22,7 +22,7 @@
   @media not all and (-moz-os-version: windows-vista) {
     @media not all and (-moz-os-version: windows-win7) {
       @media not all and (-moz-os-version: windows-win8) {
-        #tabbrowser-tabs[tabmix_currentStyle~="bg"] > .tabbrowser-tab[visuallyselected="true"] > .tab-stack > .tab-background > .tab-background-middle {
+        #tabbrowser-tabs[tabmix_currentStyle~="bg"]  .tabbrowser-tab[visuallyselected="true"] > .tab-stack > .tab-background > .tab-background-middle {
```
Comment (onemen) [inline addon/chrome/skin/app_version/78/win/tabStyles.css:25]: Why did you remove > between ```#tabbrowser-tabs``` and ```.tabbrowser-tabs```
```#tabbrowser-tabs[tabmix_currentStyle~="bg"] > .tabbrowser-tab```

Doesn't  tabbrowser-tab direct child of tabbrowser-tabs? 
It is better for specificity with parent > child.
in some places you replace .tabbrowser-tab class with id #tabbrowser-tab.
I think we should be consistent

This comment apply to all other changes.

File: addon/chrome/content/tab/tabBindings.js [hunk 2]
```diff
@@ -224,7 +224,7 @@
       if (aTab.mIsHover && aTab.getAttribute("showbutton") != "on" &&
         !aTab.hasAttribute("dragged")) {
         if (TabmixTabbar.widthFitTitle)
-          aTab.style.setProperty("width", Tabmix.getBoundsWithoutFlushing(aTab).width + "px", "important");
+          aTab.style.setProperty("max-width", Tabmix.getBoundsWithoutFlushing(aTab).width + "px", "important");
```
Comment (onemen) [inline addon/chrome/content/tab/tabBindings.js:227]: Why max-width ?
everywhere else i did use ```style.setProperty("width"....```

File: addon/modules/DynamicRules.jsm [hunk 3]
```diff
@@ -29,7 +29,7 @@ XPCOMUtils.defineLazyGetter(this, "isMac", () => {
   return TabmixSvc.isMac && !TabmixSvc.isPaleMoon;
 });
 
-const NAMESPACE = '@namespace url("http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul");\n';
+//const NAMESPACE = '@namespace url("http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul");\n'; //set namespace now preventing style been apply to elements which are nolonger of xul. pending to remove
```
Comment (onemen) [inline addon/modules/DynamicRules.jsm:32]: I did wonder about this 👍

File: addon/chrome.manifest [hunk 4]
```diff
@@ -36,9 +36,7 @@ locale    tabmixplus   zh-TW         chrome/locale/zh-TW/
 
 skin      tabmixplus   classic/1.0   chrome/skin/
 
-skin      tabmix-version tabmixplus  chrome/skin/app_version/39.0/       appversion>=39.0a1
-skin      tabmix-version tabmixplus  chrome/skin/app_version/60.0/       appversion>=60.0a1
-skin      tabmix-os    classic/1.0   chrome/skin/app_version/41.0/win/   os=WINNT appversion>=41.0a1
-skin      tabmix-os    classic/1.0   chrome/skin/app_version/60.0/win/   os=WINNT appversion>=60.0a1
+skin      tabmix-version tabmixplus  chrome/skin/app_version/78/       appversion>=78
```
Comment (onemen) [inline addon/chrome.manifest:39]: Why 78?
Is it for Waterfox

File: addon/chrome/skin/app_version/78/linux/browser.css [hunk 5]
```diff
@@ -7,12 +7,12 @@
 
 @import url("chrome://tabmixplus/skin/app_version/all/linux/browser.css");
 @import url("chrome://tabmixplus/skin/app_version/all/win/progress.css");
-@import url("chrome://tabmixplus/skin/app_version/39.0/tabStyles.css");
+@import url("chrome://tabmixplus/skin/app_version/86/tabStyles.css");
```
Comment (onemen) [inline addon/chrome/skin/app_version/78/linux/browser.css:10]: why not 78 here ?

**Review by onemen — CHANGES_REQUESTED** [review]

File: addon/chrome/skin/app_version/78/win/tabStyles.css (hunk 1 above)
Comment (117649) [reply addon/chrome/skin/app_version/78/win/tabStyles.css:25]: Not any more. Now there is other things between  `#tabbrowser-tabs` and `.tabbrowser-tabs`

I only replace things like `.tabbrowser-arrowscrollbox` to `#tabbrowser-arrowscrollbox` this basically matching changes of firefox.

File: addon/chrome/content/tab/tabBindings.js (hunk 2 above)
Comment (117649) [reply addon/chrome/content/tab/tabBindings.js:227]: I've tested 'width' has no effect here anymore tab width changes regardless.

File: addon/chrome.manifest (hunk 4 above)
Comment (117649) [reply addon/chrome.manifest:39]: The `appversion>=xx` means this rule apply to versions above xx.
78 is the last ESR version, and yes WF current is base on that version.
The firefox UI structure stabled at what we see today back to about 73, I think set minimum support version to 78 is reasonable enough.

File: addon/chrome/skin/app_version/78/linux/browser.css (hunk 5 above)
Comment (117649) [reply addon/chrome/skin/app_version/78/linux/browser.css:10]: I was trying use 86 first but as I state above it not yet come and with `appversion>=86` it seems asking for trouble.
This is a place I missing out will update this.

File: addon/chrome/skin/app_version/78/win/tabStyles.css (hunk 1 above)
Comment (onemen) [reply addon/chrome/skin/app_version/78/win/tabStyles.css:25]: OK, bet lets follow Firefox convention:
replace ```#tabbrowser-tabs[tabmix_currentStyle~="bg"] > .tabbrowser-tab[visuallyselected="true"]```
with ```#tabbrowser-tabs[tabmix_currentStyle~="bg"] > #tabbrowser-arrowscrollbox > .tabbrowser-tab[visuallyselected="true"]```

File: addon/chrome/content/tab/tabBindings.js (hunk 2 above)
Comment (onemen) [reply addon/chrome/content/tab/tabBindings.js:227]: I have to test it more, i'v notice that that tab width preference not working properly.
if it true that we need to change it everywhere in the code that handle tab width when TabmixTabbar.widthFitTitle is true
lets move this issue to different PR.

just remove this change from here

Comment (117649) [reply addon/chrome/content/tab/tabBindings.js]: Ok, it looks like this is not the real problem I have some styles relate to width fit title left not been fixed, once fixed those 'width' seems functioning fine.

File: addon/chrome/skin/app_version/78/win/tabStyles.css (hunk 1 above)
Comment (117649) [reply addon/chrome/skin/app_version/78/win/tabStyles.css:25]: This have been done.

Comment (117649) [reply addon/chrome/skin/app_version/78/win/tabStyles.css:25]: This have been done.

**Review by onemen — CHANGES_REQUESTED** [review]
fix these lines as well
```
7 results - 2 files

addon\chrome\skin\app_version\78\linux\browser.css:
  12  /***  Tab close button when user set background color ***/
  13: #TabsToolbar[brighttext] > #tabbrowser-tabs[tabmix_unloadedStyle~="bg"]  .tabbrowser-tab:not([visuallyselected="true"])[tabmix_tabState="unloaded"]:not(:hover) .tab-close-button,
  14: #TabsToolbar[brighttext] > #tabbrowser-tabs[tabmix_unreadStyle~="bg"]  .tabbrowser-tab:not([visuallyselected="true"])[tabmix_tabState="unread"]:not(:hover) .tab-close-button,
  15: #TabsToolbar[brighttext] > #tabbrowser-tabs[tabmix_otherStyle~="bg"]  .tabbrowser-tab:not([visuallyselected="true"]):not([tabmix_tabState]):not(:hover) > .tab-close-button {
  16    /*noinspection CssInvalidFunction*/

addon\chrome\skin\app_version\78\win\tabStyles.css:
  11  #main-window[tabsintitlebar][sizemode=normal] #toolbar-menubar[autohide="true"][inactive] + #TabsToolbar
  12:     #tabbrowser-tabs[tabmix_unloadedStyle~="aus"]  .tabbrowser-tab:not([visuallyselected="true"])[tabmix_tabState="unloaded"] > .tab-stack > .tab-background > .tab-background-middle,
  13  #main-window[tabsintitlebar][sizemode=normal] #toolbar-menubar[autohide="true"][inactive] + #TabsToolbar
  14:     #tabbrowser-tabs[tabmix_unreadStyle~="aus"]  .tabbrowser-tab:not([visuallyselected="true"])[tabmix_tabState="unread"] > .tab-stack > .tab-background > .tab-background-middle,
  15  #main-window[tabsintitlebar][sizemode=normal] #toolbar-menubar[autohide="true"][inactive] + #TabsToolbar
  16:     #tabbrowser-tabs[tabmix_otherStyle~="aus"]  .tabbrowser-tab:not([visuallyselected="true"]):not([tabmix_tabState]) > .tab-stack > .tab-background > .tab-background-middle {
  17    clip-path: none !important;

  24        @media not all and (-moz-os-version: windows-win8) {
  25:         #tabbrowser-tabs[tabmix_currentStyle~="bg"]  .tabbrowser-tab[visuallyselected="true"] > .tab-stack > .tab-background > .tab-background-middle {
  26            background-size: auto 100%, auto 100%, auto 100%;
```

Change all comment to 78
```
6 results - 6 files

addon\chrome\skin\app_version\78\preferences.css:
  1  /**
  2:     Firefox version 60+
  3  

addon\chrome\skin\app_version\78\linux\browser.css:
  1  /**
  2:     Firefox version 60+
  3  

addon\chrome\skin\app_version\78\linux\preferences.css:
  1  /**
  2:     Firefox version 60+
  3  

addon\chrome\skin\app_version\78\mac\browser.css:
  1  /**
  2:     Firefox version 60+
  3  

addon\chrome\skin\app_version\78\mac\preferences.css:
  1  /**
  2:     Firefox version 60+
  3  

addon\chrome\skin\app_version\78\win\preferences.css:
  1  /**
  2:     Firefox version 60+
  3  

2 results - 2 files

addon\chrome\skin\app_version\78\tabStyles.css:
  1  /**
  2:     Firefox version 39+
  3      Bug 1066531 replaced selected attribute with visuallyselected

addon\chrome\skin\app_version\78\win\tabStyles.css:
  1  /**
  2:     Firefox version 39+
  3      Bug 1066531 replaced selected attribute with visuallyselected

```

**Review by onemen — APPROVED** [review]

**Comment by onemen** [comment]
Looking good I'm going to merge it now, thank you for you hard work 🥇
